### PR TITLE
sort joint names alphabetically in yaml

### PR DESCRIPTION
**Pull request by RutgerVanBeek**
## Description

Lately the sorting of the joints was only done before uploading them to the parameter server (#252). I thought this had no impact, but the joint names are in a different order in the `ImcState.msg` and `AfterLimitJointCommand`. This is a bit unfortunate for the RQT multiplot configurations and the conversion to csv. The easiest was to make sure the joints are used alphabetically again. I tried implementing the sort in the hardware builder, but the happens in place and because of the `unique_ptr` I didn't get it to work. Sorting in the yaml was easier. 
## Changes

* Sort the joint names in the yaml alphabetically

### CODE REVIEW

**Review by Joris-JW — DISMISSED** [review]
nice.

File: march_hardware_builder/robots/march4.yaml [hunk 1]
```diff
@@ -107,17 +77,47 @@ march4:
...
+        temperatureges: {slaveIndex: 9, byteOffset: 4}
+        imotioncube:
+          slaveIndex: 11
+          absoluteEncoder:
+            resolution: 17
+            minPositionIU: 41547
+            maxPositionIU: 85287
+          incrementalEncoder:
+            resolution: 12
+            transmission: 101
```
Comment (Ishadijcks) [inline march_hardware_builder/robots/march4.yaml]: Having no new line can break the parsing of YAML files

File: march_hardware_builder/robots/march4.yaml [hunk 2]
```diff
@@ -2,17 +2,17 @@ march4:
   ifName: enp2s0
   ecatCycleTime: 4
   joints:
```
Comment (Ishadijcks) [inline march_hardware_builder/robots/march4.yaml:5]: Add a comment here that new joints have to be sorted alphabetically

**Review by Ishadijcks — CHANGES_REQUESTED** [review]
Requesting changes as this could introduce some hard to find bugs. See comments

Must this also be added to other robots (`march3.yaml`) for compatibility?

**Comment by RutgerVanBeek** [comment]
> Requesting changes as this could introduce some hard to find bugs. See comments
> 
> Must this also be added to other robots (`march3.yaml`) for compatibility?

No, because we have no rqt multiplot configurations for the march3. It is not necessarily that it should be alphabetically, but that is what some (non-essential) things assume, liike the rqt muliplot configurations.

**Review by Olavhaasie — APPROVED** [review]

**Review by Joris-JW — APPROVED** [review]

**Pull request merged**